lib/get-api.js

```diff
@@ -54,17 +54,18 @@ function getApi (state) {
     connectionStatus: hoodieConnectionStatus,
 
     // helpers
-    request: require('./request').bind(this, state),
+    request: require('./request').bind(null, state),
     log: log,
-    plugin: require('./plugin'),
 
     // events
-    on: require('./events').on.bind(this, state),
-    one: require('./events').one.bind(this, state),
-    off: require('./events').off.bind(this, state),
-    trigger: require('./events').trigger.bind(this, state)
+    on: require('./events').on.bind(null, state),
+    one: require('./events').one.bind(null, state),
+    off: require('./events').off.bind(null, state),
+    trigger: require('./events').trigger.bind(null, state)
   }
 
+  api.plugin = require('./plugin').bind(null, api, state)
```

Comment:
as mentiond above, we try to omit `this` in general. But we need to have access to the `api` that we export as the plugin method must be able to alter it.